[Improvement]: Allow filter definition's page limit to be int as well

dolmit-tanel-paaro opened the pull request:
Resolves https://github.com/pimcore/ecommerce-framework-bundle/issues/195

dolmit-tanel-paaro committed:
allow integers as filter definition's page limit value

fashxp commented:
nice, do we need a migration there for existing data?

dolmit-tanel-paaro commented:
No, existing decimal values will get converted to integers, meaning nothing really happens to the values.  
But everyone will have to update their `FilterDefinition` class definition to support the new return typehint.

fashxp commented:
> But everyone will have to update their FilterDefinition class definition to support the new return typehint.

That's also what I meant with migration :-). At least we will need an upgrade note. But I'm not sure though, how a migration path could look like? 

Automatically overwriting the class definition is dangerous, because there might custom adaption to that class definition. 
Before the upgrade you won't be able to manually change the typehint, because then they won't match to the old abstract classes anymore and break the system. 
After a composer update the system also will be broken for the same reason - the typehints do not match to the abstract classes anymore. 

Any ideas?

dolmit-tanel-paaro commented:
Yeah, this situation is just like the one we faced when we migrated from Pimcore 6 to Pimcore X. Everything was broken because of return typehints because they were made mandatory.  

Back then we solved it by upgrading to Pimcore X and then manually go over our class definitions and update return typehints manually.  

The migration plan:
* update the package
* fix the issue(s) manually
* run `composer` again

fashxp commented:
Hmm, we could do a two step process.
for now, we just change return-type of interface to `float|int|null`, which won't break any existing installations and also does not require any changes on existing data - but allows safely to convert the data over time. 

And then with next major, we could switch to pure int. 

The change in `src/Resources/install/class_sources/class_FilterDefinition_export.json` we could keep, as it just reflects for new installations.

dolmit-tanel-paaro commented:
Yeah, that's a good approach. I'll make the changes.

dolmit-tanel-paaro commented:
Updated abstract class, allowing float and int values.
Added todo for next major.
[diff](https://github.com/pimcore/ecommerce-framework-bundle/compare/2d4e4d58de0bd6610a0d7e67be745db3dce45258..a6bb19972cac8dbfe864d874369fc88b9429e963)

fashxp commented:
thx very much
